src/frontend/AppContainerWrapper.tsx

```diff
@@ -92,13 +99,42 @@ function hidePracticeModeTemporarily(route: string | null) {
   return route !== null && TEMP_HIDE_PRACTICE_MODE_UI.includes(route);
 }
 
+const AuthContainer = createAppContainer(AuthStack);
+
 const AppContainerWrapper = () => {
   const navRef = React.useRef<NavigationContainerComponent>();
   const [inviteModalEnabled, setInviteModalEnabled] = React.useState(true);
   const [queuedInvite, setQueuedInvite] = React.useState<string | null>(null);
   const [{ experiments }] = React.useContext(SettingsContext);
   const [hidePracticeBar, setHidePracticeBar] = React.useState(true);
   const [hidePracticeMode, setHidePracticeMode] = React.useState(false);
+  const [{ authStatus }, setAuthState] = React.useContext(SecurityContext);
+  const isMounted = React.useRef(false);
+
+  React.useEffect(() => {
+    isMounted.current = true;
+  }, []);
+
+  React.useEffect(() => {
+    const appStateListener = AppState.addEventListener(
+      "change",
+      handleStateAndSetCheckPassFlag
```

Comment:
may have already commented this before, but would move the function definition into this effect because:

1. it's not used elsewhere in the component
2. it's missing in the effects dep list, so just easier to avoid remembering to do that

Reply:
even simpler would be to inline the function i.e.:

```ts
  React.useEffect(() => {
    const appStateListener = AppState.addEventListener(
      "change",
      (nextAppState: AppStateStatus) => {
        if (authStatus !== "notRequired") {
          if (nextAppState === "inactive" || nextAppState === "background") {
            setAuthState({ type: "setAuthStatus", newAuthStatus: "pending" });
          }
        }
      }
    );

    return () => appStateListener.remove();
  }, [authStatus]);
```